service/controller/resource/clusterconfigmap/current.go

```diff
@@ -31,29 +32,25 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) ([]*cor
 		return nil, nil
 	}
 
-	// The cluster config map is deleted implicitly by the provider operator when
-	// it deletes the tenant cluster namespace in the control plane.
-	if key.IsDeleted(&cr) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("not deleting config map %#q for tenant cluster %#q", key.ClusterConfigMapName(&cr), key.ClusterID(&cr)))
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-		resourcecanceledcontext.SetCanceled(ctx)
-		return nil, nil
-	}
```

Comment:
I think this is no longer correct as the namespace is now managed by the cpnamespace resource.

@xh3b4sd Can you confirm?

Reply:
I am not sure about this. The logic did not change. It only moved. Question is if this breaks anything.

Reply:
It's there because in legacy the provider operator creates and deletes the namespace. AIUI this is now done here by the `cpnamespace` resource.

But I can put back the logic and just fix the comments? As its no longer accurate. WDYT?

Reply:
Undecided. I am not sure what the procedure should be. I think that resources should all process their own cleanup mechanism. I feel like deferring to the namespace termination was the lazy approach. This is maybe something we should get a clearer idea about for the future. I put a card into the SIG Operator Board. When your PR works let's go with it. I think it would be the cleaner and more proper solution. When you run into problems, let's keep the cancellation here and move on.